client-common/src/tendermint/rpc_client/sync_rpc_client.rs

```diff
@@ -11,20 +11,29 @@ use tendermint::{lite, validator};
 use tokio::runtime::Runtime;
 
 use chain_core::state::ChainState;
+use std::sync::Mutex;
 
 use super::async_rpc_client::AsyncRpcClient;
 use crate::{
     tendermint::{lite::TrustedState, types::*, Client},
     Error, ErrorKind, Result, ResultExt,
 };
 
+use futures_util::sink::SinkExt;
+use tokio_tungstenite::tungstenite::protocol::frame::coding::CloseCode;
+use tokio_tungstenite::tungstenite::protocol::frame::CloseFrame;
+use tokio_tungstenite::tungstenite::Message;
+
 const RESPONSE_TIMEOUT: Duration = Duration::from_secs(10);
 
 /// Wraps asynchronous RPC client and executes it in tokio runtime
 #[derive(Clone)]
 pub struct SyncRpcClient {
-    runtime: Arc<Runtime>,
-    async_rpc_client: AsyncRpcClient,
+    runtime: Arc<Mutex<Runtime>>,
+    /// ASYNC RPC CLIENT
+    pub async_rpc_client: AsyncRpcClient,
+    /// graceful close on Drop
+    pub disconnect_on_drop: bool,
```

Comment:
Why do we need a flag for this? Shouldn't this be the default behaviour?

Reply:
RpcClient is cloned, without that, integration test breaks.
also don't to shutdown it gracefully for all cases